ruby/lib/gherkin3/token_scanner.rb

```diff
@@ -11,15 +13,23 @@ module Gherkin3
   # to look for Gherkin keywords for the associated language. The keywords are defined 
   # in gherkin-languages.json.
   class TokenScanner
+    protected
+
+    attr_reader :io
```

Comment:
Additionally:
The scanner raises an error to make the user aware about the content it accepts.